TACC-Cloud/geoapi
# Task/WG-383: add queue for heavy tasks

## Overview: ##

Add `heavy` queue for computationally intensive tasks. In this PR, the queue will be used by:
* Point cloud processing, which takes a lot of CPU and memory when processing large point clouds.  
* The street view uploading task can take a long time (even though it doesn't use a lot of CPU or memory).

The number of workers for both queues is:
*  `default` queue uses Celery's default behavior: nproc number of worker processes.
* `heavy` queue is set to nproc/2 number of workers (so intensive/long tasks don't monopolize things)
 
Note: While we've discussed categorizing tasks based on their true intensity/need (e.g., differentiating between small and large point cloud processing), this PR doesn't implement that. Instead, it makes incremental improvements by just having a smaller queue for large tasks to i) reduce the risk of memory exhaustion during large point cloud processing, and ii) prevent system monopolization by intensive tasks.

## Related Jira tickets: ##

* [WG-383](https://tacc-main.atlassian.net/browse/WG-383)

## Summary of Changes: ##

## Testing Steps: ##
1. Run and ensure you load features like images and then also point clouds to ensure both queues are working.
